packages/@aws-cdk/aws-redshift/lib/cluster-parameter-change-reboot-handler/index.ts

```diff
@@ -0,0 +1,53 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Redshift } from 'aws-sdk';
+
+const redshift = new Redshift();
+
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent): Promise<void> {
+  if (event.RequestType !== 'Delete') {
+    return rebootClusterIfRequired(event.ResourceProperties?.ClusterId, event.ResourceProperties?.ParameterGroupName);
+  } else {
+    return;
+  }
+}
+
+async function rebootClusterIfRequired(clusterId: string, parameterGroupName: string): Promise<void> {
+  return executeActionForStatus(await getApplyStatus());
+
+  // https://docs.aws.amazon.com/redshift/latest/APIReference/API_ClusterParameterStatus.html
+  async function executeActionForStatus(status: string, retryDurationMs?: number): Promise<void> {
+    await sleep(retryDurationMs ?? 0);
+    if (['pending-reboot', 'apply-deferred', 'apply-error', 'unknown-error'].includes(status)) {
```

Comment:
why is `unknown-error` in this list? If the error is something that isn't solved by a reboot, then the custom resource will just continuously reboot the cluster, no?

Reply:
That's a good point, I'll remove it. Anecdotally I had an `unknown error` that was resolved by a reboot, but since `unknown errors` encompass an infinite set of errors the situation you pointed out could very easily happen